File: src/Application Specific/Classes/Entity Collections/Spares/ProductTransaction/ProductTransactionValidator.vb
Namespace Entity

Namespace ProductTransactions

Public Class ProductTransactionValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oProductTransaction As ProductTransaction)

        'make sure that contact object is valid
        If oProductTransaction.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oProductTransaction.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/ProductTransaction/ProductTransactionValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ProductTransactions
    {
        // make sure that contact object is valid
        public class ProductTransactionValidator : BaseValidator
        {
            public void Validate(ProductTransaction oProductTransaction)
            {
                // make sure that contact object is valid
                if (oProductTransaction.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oProductTransaction.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}